lib/main.dart

```diff
@@ -28,29 +31,58 @@ void main() {
   } else {
     ArcGISEnvironment.apiKey = apiKey;
   }
-  runApp(const MyApp());
+  runApp(const SampleViewerApp());
 }
 
-class MyApp extends StatelessWidget {
-  const MyApp({super.key});
+class SampleViewerApp extends StatefulWidget {
+  const SampleViewerApp({super.key});
+
+  @override
+  State<SampleViewerApp> createState() => _SampleViewerAppState();
+}
+
+class _SampleViewerAppState extends State<SampleViewerApp> {
+  final _samples = <Sample>[];
+  bool _ready = false;
+
+  @override
+  void initState() {
+    loadSamples();
+    super.initState();
+  }
 
   @override
   Widget build(BuildContext context) {
     final ColorScheme colorScheme =
         ColorScheme.fromSeed(seedColor: Colors.deepPurple);
+    const title = 'ArcGIS Maps SDK for Flutter Samples';
 
     return MaterialApp(
-      title: 'ArcGIS Maps SDK for Flutter Samples',
+      title: title,
       theme: ThemeData(
         colorScheme: colorScheme,
         appBarTheme: AppBarTheme(backgroundColor: colorScheme.inversePrimary),
       ),
       home: Scaffold(
         appBar: AppBar(
-          title: const Text('ArcGIS Maps SDK for Flutter Samples'),
+          title: const Text(title),
         ),
-        body: const SampleListView(),
+        body: _ready
+            ? SampleListView(samples: _samples)
+            : const Center(child: Text('Loading samples...')),
       ),
     );
   }
+
+  void loadSamples() async {
+    final jsonString =
+        await rootBundle.loadString('assets/generated_samples_list.json');
+    Map<String, dynamic> sampleData = jsonDecode(jsonString);
```

Comment:
I think `Map<String, dynamic> sampleData` can just be `final sampleData`?

Reply:
So if I don't define `Map<String, dynamic>` it's just type `dynamic` being returned. Which works, but I guess I naturally feel more comfortable defining the type there. What do you think?